bufferevent_sock.c

```diff
@@ -498,6 +498,41 @@ bufferevent_connect_getaddrinfo_cb(int result, struct evutil_addrinfo *ai,
 }
 
 int
+bufferevent_socket_set_src_address(struct bufferevent *bev,
+    const struct sockaddr *addr, int socklen)
+{
+	struct bufferevent_private *bufev_p =
+	    EVUTIL_UPCAST(bev, struct bufferevent_private, bev);
+
+	evutil_socket_t fd;
+	int r = 0;
+	int result = -1;
+
+	bufferevent_incref_and_lock_(bev);
+
+	if (!bufev_p || !addr)
+		goto done;
+
+	/* do we need to check to see if the socket is connecting/connected
+	 * already?
+	 */
+
+	fd = bufferevent_getfd(bev);
+	if (fd < 0)
+		goto done;
+
+	r = evutil_socket_bind_address(fd, addr, socklen);
+	if (r < 0)
+		goto done;
+
+	result = 0;
+
+done:
+	bufferevent_decref_and_unlock_(bev);
+	return result;
+}
```

Comment:
Well, I've renamed this to `bufferevent_socket_bind()` from `bufferevent_socket_set_src_addr()`... so this has to call `evutil_socket_bind()` to get the underlying `bind()` to happen.